control-plane/pkg/reconciler/base/reconciler.go

```diff
@@ -108,25 +108,27 @@ func isAtLeastOneRunning(pods []*corev1.Pod) bool {
 
 type ConfigMapOption func(cm *corev1.ConfigMap)
 
+func NoopConfigmapOption(cm *corev1.ConfigMap) {}
+
 func (r *Reconciler) GetOrCreateDataPlaneConfigMap(ctx context.Context, options ...ConfigMapOption) (*corev1.ConfigMap, error) {
 
 	cm, err := r.KubeClient.CoreV1().
 		ConfigMaps(r.DataPlaneConfigMapNamespace).
 		Get(ctx, r.DataPlaneConfigMapName, metav1.GetOptions{})
 
 	if apierrors.IsNotFound(err) {
-		cm, err = r.createDataPlaneConfigMap(ctx, options)
+		cm, err = CreateDataPlaneConfigMap(ctx, r.KubeClient, r.DataPlaneConfigMapNamespace, r.DataPlaneConfigMapName, options...)
```

Comment:
I think we should transform existing ConfigMaps with the options for consistency.

Reply:
I think this captured the general idea: https://github.com/knative-sandbox/eventing-kafka-broker/pull/2361#discussion_r928702545